### ISSUE TITLE
default country with router.locale

### ISSUE THREAD

**Issue by viccabezos** [goveo/react-international-phone#61]
**Describe the bug**
first i a not sure it's a bug, but i am not sure it's not something that already exists  so it's more of a question 🙏🏼 

i am trying to set the default language dynamically using the router, maybe the way i am doing it is not the best way as i am having an error of an undetected iso2

**To Reproduce**

CODE

````
export function PhoneField(props: { required?: boolean }) {
  const { field, error } = useTsController<string>();
  const { label } = useDescription();
  const router = useRouter();

  console.log('router', router.locale, router.defaultLocale);
  const defaultCountryRouter = ((router.locale ?? router.defaultLocale) as CountryIso2) || 'fr';
  console.log('defo', defaultCountryRouter);

  return (
    <>
      <Label label={label} error={error} required={props.required} />
      <PhoneInput
        defaultCountry={defaultCountryRouter}
        value={field.value}
        onChange={field.onChange}
        className={cn('rounded', {
          'ring-1 ring-red-500': error?.errorMessage,
        })}
        inputClassName={inputClassNames}
      />
      <ErrorMessage {...error} />
    </>
  );
}
```

ERROR: Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'iso2')


**Expected behavior**
default country should be settled with the router

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**

- OS: [e.g. iOS] macbook
- Browser [e.g. chrome, safari] chrome, arc
- Version Version 114.0.5735.198 (Build officiel) (x86_64)

**Comment by goveo** [comment]
@viccabezos Hey, If I understood correctly you are using a Next.js router. Seems like `router.defaultLocale` returning a string in a format like `fr-FR`. `defaultCountry` prop supports country codes in a format like `fr`.
I think you can get the county code from `router.defaultLocale` like this:
```ts
const countryCode = router.defaultLocale.split('-')[0];
```
Let me know if it helps.

**Comment by goveo** [comment]
I gonna fix the `TypeError: Cannot read properties of undefined (reading 'iso2')` in the scope of this issue.
If a wrong country code is provided it should be set to "us" by default I think (and show the error in the console)

**Comment by viccabezos** [comment]
@goveo  Hello, thanks for your answers, 
Yes you did understood correctly sorry for not precising earlier, i've tried this but i have the same TS error, i'll keep in mind to check when a fix is released to try again txs :)

**Comment by goveo** [comment]
@viccabezos Error should be fixed in [v2.3.1](https://github.com/goveo/react-international-phone/releases/tag/v2.3.1) United States will be set if an invalid country is provided to the `defaultCountry` prop.
Please let me know if it helps :)
